### ISSUE TITLE
Dynamic dispatch postfixes

### ISSUE THREAD

**Issue by ryzhyk** [feldera/feldera#1518]
The dynamic dispatch PR leaves some tasks unfinished:

- [ ] **Move storage back into a separate crate.** This requires separating the dynamic dispatch infra (the `/dynamic`) module into its own crate, which will become a common dependency for `dbsp`  and `storage`. The main blocker for this is the `CompositeSerializer` type, which is declared in `storage` and is used to define serialization traits in `dynamic`, thus introducing a circular dependency between `storage` and `dynamic`. @gz , @blp , I may need your help with this.
- [x] **Rename file-base batches and traits.** When working on dynamic dispatch, I changed naming conventions for batches and layers. I did not apply them to file-based batches in order not to further complicate the rebasing (or merging) of @blp 's oustanding PRs. Once these PR's have been merged, we should do the renaming.
- [ ] **Fix up ldbc and fraud benchmarks.** I ran into some complications adapting these benchmarks to the new API and simply disabled them.
- [ ] **Fix safety annotations.** Several traits defined in `dynamic` have safe signatures, however their safety invariants are only enforced in debug builds, i.e., they are unsafe in reality. Making them `unsafe` will propagate to many other places, but that's the correct thing to do, so we should do it.

**Comment by blp** [comment]
I don't think we need to move storage into a different crate. (We do have the ability now to implement backends in other crates.)
